src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c

```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+        case 0: /* Have not talked to Shikashi as of entering observatory last */
+        case 1: /* Have finished a conversation with Shikashi */
+            switch (gSaveContext.playerForm) {
+                case PLAYER_FORM_DEKU:
+                    if ((gSaveContext.weekEventReg[17] & 0x10) != 0) {
+                        if ((gSaveContext.weekEventReg[74] & 0x80) != 0) {
+                            this->textId = 0x5F4;
+                        } else {
+                            this->textId = 0x5E2;
+                        }
+                    } else {
+                        this->textId = 0x5EC;
+                        gSaveContext.weekEventReg[17] |= 0x10;
+                    }
+                    break;
+                case PLAYER_FORM_HUMAN:
+                    if (Player_GetMask(globalCtx) == PLAYER_MASK_KAFEIS_MASK) {
+                        this->textId = 0x236A;
+                    } else if ((gSaveContext.weekEventReg[74] & 0x10) != 0) {
+                        this->textId = 0x5F6;
+                    } else {
+                        this->textId = 0x5F5;
+                        gSaveContext.weekEventReg[74] |= 0x10;
+                    }
+                    break;
+                case PLAYER_FORM_GORON:
+                case PLAYER_FORM_ZORA:
+                    if ((gSaveContext.weekEventReg[75] & 8) != 0) {
+                        this->textId = 0x5E4;
+                    } else {
+                        this->textId = 0x5DC;
+                        gSaveContext.weekEventReg[75] |= 8;
+                    }
+                    break;
+            }
+            break;
+        case 3: /* Currently engaged in post-telescope Shikashi dialogue */
+            switch (gSaveContext.playerForm) {
+                case PLAYER_FORM_DEKU:
+                    if ((gSaveContext.weekEventReg[74] & 0x80) != 0) {
+                        this->textId = 0x5F2;
+                    } else {
+                        this->textId = 0x5F1;
+                    }
+                    func_800B8500(&this->actor, globalCtx, this->actor.xzDistToPlayer, this->actor.yDistToPlayer, 0);
+                    break;
+                case PLAYER_FORM_HUMAN:
+                    this->textId = 0x5F7;
+                    break;
+                case PLAYER_FORM_GORON:
+                case PLAYER_FORM_ZORA:
+                    switch (CURRENT_DAY) {
+                        case 1:
+                            this->textId = 0x5E9;
+                            break;
+                        case 2:
+                            this->textId = 0x5EA;
+                            break;
+                        case 3:
+                            tempDay = gSaveContext.day;
+                            tempTimeBeforeMoonCrash =
+                                ((-(tempDay % 5 << 0x10) - ((u16)(gSaveContext.time - 0x4000))) + 0x40000);
+                            if (tempTimeBeforeMoonCrash < 2730.6667f /* 1 hr */) {
```

Comment:
wtf?
Errr... well, if it matches...
Hopefully we could rewrite this in a more readable way eventually. There's no need for you to do it.

Reply:
```suggestion
                        case 3:
                            tempDay = gSaveContext.day;
                            tempTimeBeforeMoonCrash =
                                ((-(tempDay % 5 << 0x10) - ((u16)(gSaveContext.time - 0x4000))) + 0x40000);
                            if (tempTimeBeforeMoonCrash < 2730.6667f) { /* 1 hr */)
```
Would rather the comment was outside.